tests/CompileTest.hs

```diff
@@ -46,18 +46,15 @@ withinRange'CH v r =
 
 setupHOME :: IO ()
 setupHOME = do
-  mhome <- lookupEnv "HOME"
-  case mhome of
-    Just home -> do
-      exists <- doesDirectoryExist home
-      when (not exists) createHOME
-    Nothing -> createHOME
+  home <- getHomeDirectory
+  exists <- doesDirectoryExist home
+  when (not exists) createHOME
 
 createHOME :: IO ()
 createHOME = do
-  tmp <- fromMaybe "/tmp" <$> lookupEnv "TMPDIR"
+  tmp <- getTemporaryDirectory
   let home = tmp </> "compile-test-home"
-  _ <- rawSystem "rm" ["-r", home]
+  removeDirectoryRecursive home
   createDirectory home
   setEnv "HOME" home
```

Comment:
Since you're using getHomeDirectory above but `setEnv "HOME"` here this is unlikely to be portable.